packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -534,6 +541,91 @@ export class CdkToolkit {
     return stacks;
   }
 
+  private stackHasNestedStacks(rootStack: cxapi.CloudFormationStackArtifact) {
+    for (const resourceName in rootStack.template.Resources) {
+      if (rootStack.template.Resources[resourceName].Type === 'AWS::CloudFormation::Stack') {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private async findPhysicalNameForNestedStack(logicalId: string, parentStackName: string | undefined, sdk: ISDK): Promise<string | undefined> {
+    if (!parentStackName) {
+      return undefined;
+    }
+    const stackResources = await this.listNestedStackResources(parentStackName, sdk);
+    return stackResources.find(sr => sr.LogicalResourceId === logicalId)?.PhysicalResourceId;
+  }
+
+  private async listNestedStackResources(parentStackName: string, sdk: ISDK): Promise<CloudFormation.StackResourceSummary[]> {
+    const ret = new Array<CloudFormation.StackResourceSummary>();
+    let nextToken: string | undefined;
+    do {
+      const stackResourcesResponse = await sdk.cloudFormation().listStackResources({
```

Comment:
This is hiding a bug - if we Stack we are diffing doesn't exist yet (we ran `cdk diff` before the first `cdk deploy`), this will fail. We need to handle that case (it also needs to be unit tested).